Buildings/Experimental/DHC/Networks/Controls/AgentPump1Pipe.mo

```diff
@@ -0,0 +1,238 @@
+   parameter Real Td = 0.1 "Time constant of derivative block";
+   parameter Modelica.Units.SI.TemperatureDifference dToff(min=0.1) = 1
+    "Temperature offset to account for heat exchanger pinch point";
+   parameter Real uLowHea = 1 "if y=true and u<uLow, switch to y=false";
+   parameter Real uHighHea = 2 "if y=false and u>uHigh, switch to y=true";
+   parameter Real uLowCoo = 1 "if y=true and u<uLow, switch to y=false";
+   parameter Real uHighCoo = 2 "if y=false and u>uHigh, switch to y=true";
+   parameter Real h = 0.15 "Hysteresis for net demand temperature calculation";
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput TSouIn(final unit="K",
+       displayUnit="degC") "Temperatures at the inlet of the source"
+    annotation (Placement(transformation(extent={{-160,50},{-120,90}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput TSouOut(final unit="K",
+      displayUnit="degC") "Agent supply temperature"
+    annotation (Placement(transformation(extent={{-160,-30},{-120,10}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput TSou(final unit="K",
+      displayUnit="degC") "Average temperature available at source"
+    annotation (Placement(transformation(extent={{-160,10},{-120,50}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput TretDis(final unit="K",
+      displayUnit="degC") "District return temperature"
+    annotation (Placement(transformation(extent={{-160,-70},{-120,-30}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput TsupDis(final unit="K",
+      displayUnit="degC") "Plant supply temperature"
+    annotation (Placement(transformation(extent={{-160,-100},{-120,-60}}),
+        iconTransformation(extent={{-160,-100},{-120,-60}})));
+
+  Buildings.Controls.OBC.CDL.Reals.Less NetDemBool(h=h)
+    "Net district demand boolean"
+    annotation (Placement(transformation(extent={{-102,-100},{-82,-80}})));
+  Buildings.Controls.OBC.CDL.Reals.Switch swi
+    annotation (Placement(transformation(extent={{110,-10},{130,10}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput y(
+    min=0,
+    max=1,
+    unit="1")
+    "Pump control signal"
+    annotation (Placement(transformation(extent={{140,-20},{180,20}}),
+        iconTransformation(extent={{140,-20},{180,20}})));
+  Buildings.Controls.OBC.CDL.Reals.PID conPIDHea(
+    controllerType=controllerType,
+    k=k,
+    Ti=Ti,
+    Td=Td,
+    yMax=yPumMax,
+    yMin=yPumMin) "Controller to track target heating setpoint temperature"
+    annotation (Placement(transformation(extent={{0,70},{20,90}})));
+
+  Buildings.Controls.OBC.CDL.Reals.PID conPIDCoo(
+    controllerType=controllerType,
+    k=k,
+    Ti=Ti,
+    Td=Td,
+    yMax=yPumMax,
+    yMin=yPumMin,
+    reverseActing=false)
+    "Controller to track target cooling setpoint temperature"
+    annotation (Placement(transformation(extent={{0,-80},{20,-60}})));
+
+  Buildings.Controls.OBC.CDL.Reals.Hysteresis hys(uLow=uLowCoo, uHigh=
+        uHighCoo)
```

Comment:
Look at *all* your code and make sure it is formatted properly. Code is written once but red 100s of times.

Reply:
I formatted all the updated code trying to make it more readable.